app/src/lib/title-bar-config.ts

```diff
@@ -0,0 +1,38 @@
+import { writeFile } from 'fs/promises'
+import { app } from 'electron'
+import { join } from 'path'
+import { TitleBarStyle } from '../ui/lib/title-bar-style'
+import { readFileSync } from 'fs'
+
+export type TitleBarConfig = {
+  titleBarStyle: TitleBarStyle
+}
+
+const getTitleBarConfigPath = () =>
+  join(app.getPath('userData'), '.title-bar-config')
+
+// The function has to be synchronous since we need its return value to create electron BrowserWindow
+function readTitleBarConfigFile(): TitleBarConfig {
+  let titleBarConfig: TitleBarConfig
+
+  try {
+    titleBarConfig = JSON.parse(readFileSync(getTitleBarConfigPath(), 'utf8'))
+
+    // Return if we found valid values
+    if (['native', 'custom'].includes(titleBarConfig.titleBarStyle)) {
+      return titleBarConfig
+    }
+  } catch (error) {}
+
+  // If the config file doesn't exists, or if it is corrupted, rewrite it with the defaults
+  titleBarConfig = { titleBarStyle: 'native' }
+  saveTitleBarConfigFile(titleBarConfig)
+
+  return titleBarConfig
+}
+
+export async function saveTitleBarConfigFile(config: TitleBarConfig) {
+  await writeFile(getTitleBarConfigPath(), JSON.stringify(config), 'utf8')
+}
+
+export const TITLE_BAR_CONFIG = readTitleBarConfigFile()
```

Comment:
:art: If it were me designing this module I would just export the `readTitleBarConfigFile()` and execute that function when needed, rather than export a `const` that eagerly executes it.

Reply:
Since a restart is required for changes to apply, The value would not get changed in normal circumstances and it should be safe to cache it as a constant.